cloudvolume/dasklib.py

```diff
@@ -0,0 +1,133 @@
+
+    Parameters
+    ----------
+    arr: dask.array
+        Data to store
+    vol_path: str
+        Path to the dataset layer. This should match storage's supported
+        providers.
+        e.g. Google: gs://$BUCKET/$DATASET/$LAYER/
+             S3    : s3://$BUCKET/$DATASET/$LAYER/
+             Lcl FS: file:///tmp/$DATASET/$LAYER/
+             Boss  : boss://$COLLECTION/$EXPERIMENT/$CHANNEL
+             HTTP/S: http(s)://.../$CHANNEL
+             matrix: matrix://$BUCKET/$DATASET/$LAYER/
+    resolution: Iterable of ints of length 3
+        The x, y, z voxel dimensions in nanometers
+    voxel_offset: Iterable of ints of length 3
+        The x, y, z beginning of dataset in positive cartesian space.
+    layer_type: str
+        "image" or "segmentation"
+    max_mip: int
+        Maximum mip level id.
+    compute, return_stored: see ``store()``
+    kwargs: passed to the ``cloudvolume.CloudVolume()`` function, e.g., compression options
+
+    Raises
+    ------
+    ValueError
+        If ``arr`` has ndim different that 3 or 4, or ``layer_type`` is unsupported.
+    """
+    import dask.array as da
+    if not da.core._check_regular_chunks(arr.chunks):
+        raise ValueError('Attempt to save array to cloudvolume with irregular '
+                         'chunking, please call `arr.rechunk(...)` first.')
+
+    if not layer_type:
+        if arr.dtype in (np.bool, np.uint32, np.uint64, np.uint16):
+            layer_type = 'segmentation'
+        elif np.issubdtype(arr.dtype, np.integer) or np.issubdtype(
+                arr.dtype, np.floating):
+            layer_type = 'image'
+        else:
+            raise ValueError('Unsupported layer_type for CloudVolume: %s' %
+                             layer_type)
+
+    if arr.ndim == 3:
+        num_channels = 1
+        chunk_size = arr.chunksize
+    elif arr.ndim == 4:
+        num_channels = arr.shape[-1]
+        chunk_size = arr.chunksize[:3]
+    else:
+        raise ValueError('Unsupported ndim for CloudVolume: %d' % arr.ndim)
+
+    info = CloudVolume.create_new_info(num_channels,
+                                       layer_type,
+                                       arr.dtype.name,
+                                       encoding,
+                                       resolution,
+                                       voxel_offset,
+                                       arr.shape[:3],
+                                       chunk_size=chunk_size,
+                                       max_mip=max_mip)
+    vol = CloudVolume(vol_path, info=info, progress=False, **kwargs)
+    # pylint: disable=no-member
+    vol.commit_info()
+    vol.provenance.processing = [{'method': 'dask_to_cloudvolume'}]
+    vol.commit_provenance()
+    return arr.store(vol,
+                     lock=False,
+                     compute=compute,
+                     return_stored=return_stored)
+
+
+def dask_from_cloudvolume(vol_path, chunks=None, name=None, **kwargs):
```

Comment:
In other areas of the codebase, we tend to call `vol_path` `cloudpath`. It would be good to standardize this (though the from_numpy function might diverge...).